components/markdown_image_expand/markdown_image_expand.tsx

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {useState, useEffect} from 'react';
+import {CSSTransition} from 'react-transition-group';
+import './markdown_image_expand.scss';
+
+export type Props = {
+    alt: string;
+    children: React.ReactNode;
+    isEmbedVisible: boolean;
+    postId: string;
+};
+
+type State = {
+    isExpanded: boolean;
+};
+
+const MarkdownImageExpand: React.FC<Props> = ({children, alt, isEmbedVisible}: Props) => {
+    const [state, setState] = useState<State>({
+        isExpanded: isEmbedVisible,
+    });
+    const {isExpanded} = state;
+
+    useEffect(() => {
+        setState({
+            isExpanded: isEmbedVisible,
+        });
+    }, [setState, isEmbedVisible]);
+
+    const handleToggleButtonClick = () => {
+        setState((oldState) => ({...oldState, isExpanded: !oldState.isExpanded}));
```

Comment:
Instead of just writing back to the state, this might need to write back to the redux state and local storage so that the expanded/collapsed state is persisted through a refresh

Reply:
Fixed.